### PR TITLE
New Template - cisco_ios_show_object-group

### PR DESCRIPTION
**Pull request by pkomissarov**
##### ISSUE TYPE
 - New Template Pull Request

##### COMPONENT
cisco_ios_show_object-group

##### SUMMARY
#850 
As @FragmentedPacket proposed, I applied changes:
Renamed OGR_NAME to NAME
Renamed OGR_TYPE to TYPE
Renamed NESTED_OGR to NESTED_GROUPS
Changed last line to
```suggestion
  ^. -> Error
```

### CODE REVIEW

**Comment by pkomissarov** [comment]
Hello, 
Question to Reviewers: do you plan to review/merge this pull request?  
More than two months without progress.

**Comment by FragmentedPacket** [comment]
@pkomissarov Sorry about the huge delay with reviewing. I will try and get it review within the next day.

We appreciate your patience.

File: templates/cisco_ios_show_object-group.textfsm [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Value Required,Filldown OGR_TYPE (Service|Network)
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_object-group.textfsm:1]: I'd probably just make these `TYPE/NAME`

File: templates/cisco_ios_show_object-group.textfsm [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Value Required,Filldown OGR_NAME (\S+)
+Value DESCRIPTION (.+)
+Value NESTED_OGR (\S+)
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_object-group.textfsm:4]: Can there be more than one NESTED_OGR? And maybe better served as we list?

Comment (FragmentedPacket) [reply templates/cisco_ios_show_object-group.textfsm:4]: I'd probably just spell this out as well: `NESTED_GROUPS`

File: templates/cisco_ios_show_object-group.textfsm [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+Value NETMASK (\d+\.\d+\.\d+\.\d+)
+Value PROTOCOL (\S+)
+Value PORT_MATCH (eq|neq|range|lt|gt)
+Value PORT ((?<!range\s)\S+)
+Value PORT_RANGE_START ((?<=range\s)\S+)
+Value PORT_RANGE_END (\S+)
+Value ICMP_TYPE (echo|echo-reply|administratively-prohibited|unreachable|port-unreachable|redirect|router-advertisement|router-solicitation|packet-too-big|time-exceeded|ttl-exceeded|parameter-problem)
+
+Start
+  ^(Service|Network) -> Continue.Clearall
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_object-group.textfsm:19]: We should be able to capturing everything within single entries.

Look into: https://github.com/networktocode/ntc-templates/blob/master/templates/cisco_ios_show_ip_interface.textfsm for how to accomplish this.

File: templates/cisco_ios_show_object-group.textfsm [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  ^${OGR_TYPE}\s+object\s+group\s+${OGR_NAME}\s*$$ -> Record
+  ^\s+Description\s+${DESCRIPTION}$$ -> Record
+  ^\s+group-object\s+${NESTED_OGR}\s*$$ -> Record
+  ^\s+(host\s+${HOST}|range\s+${HOST_RANGE_START}\s+${HOST_RANGE_END}|${ANY}|${NETWORK}\s+${NETMASK})\s*$$ -> Record
+  ^\s+icmp\s+${ICMP_TYPE}\s*$$ -> Record
+  ^\s+${PROTOCOL}\s+${PORT_MATCH}\s+(${PORT_RANGE_START}\s+${PORT_RANGE_END}|${PORT})\s*$$ -> Record
+  ^\s+${PROTOCOL}\s*$$ -> Record
+  ^\s*$$
+  ^.* -> Error "Could not parse line:"
```
Comment (FragmentedPacket) [inline templates/cisco_ios_show_object-group.textfsm:28]: We typically just require ` ^. -> Error`

File: templates/cisco_ios_show_object-group.textfsm (hunk 3 above)
Comment (FragmentedPacket) [reply templates/cisco_ios_show_object-group.textfsm:19]: Along this line, you will want to remove all the `-> Record` on child lines.

**Review by FragmentedPacket — CHANGES_REQUESTED** [review]

File: templates/cisco_ios_show_object-group.textfsm (hunk 3 above)
Comment (FragmentedPacket) [reply templates/cisco_ios_show_object-group.textfsm:19]: Let me discuss internally as this is quite a different case, and need to make sure we're aligned on how we should capture the data.

Comment (FragmentedPacket) [reply templates/cisco_ios_show_object-group.textfsm:19]: Ignore this comment and I will resolve any other feedback that doesn't need to be addressed.

File: templates/cisco_ios_show_object-group.textfsm (hunk 1 above)
Comment (FragmentedPacket) [reply templates/cisco_ios_show_object-group.textfsm:1]: Can you potentially align around the naming such as similar template? https://github.com/networktocode/ntc-templates/blob/master/templates/cisco_asa_show_running-config_object_network.textfsm.

Not too many changes required to get this merged in and we appreciate your patience!

**Comment by FragmentedPacket** [comment]
@pkomissarov Just wanted to see if you could address the open conversations here and then we can get this merged in.

File: templates/cisco_ios_show_object-group.textfsm (hunk 1 above)
Comment (pkomissarov) [reply templates/cisco_ios_show_object-group.textfsm:1]:
```suggestion
Value Required,Filldown TYPE (Service|Network)
```
OK, I agree

File: templates/cisco_ios_show_object-group.textfsm (hunk 2 above)
Comment (pkomissarov) [reply templates/cisco_ios_show_object-group.textfsm:4]:
```suggestion
Value NESTED_GROUPS (\S+)
```
OK, I agree

File: templates/cisco_ios_show_object-group.textfsm (hunk 4 above)
Comment (pkomissarov) [reply templates/cisco_ios_show_object-group.textfsm:28]:
```suggestion
  ^. -> Error
```
OK

**Pull request merged**